addon/adapters/pouch.js

```diff
@@ -363,10 +359,31 @@ export default DS.RESTAdapter.extend({
     return this.get('db').rel.save(this.getRecordTypeName(type), data);
   },
 
-  deleteRecord: function (store, type, record) {
+  deleteRecord: function (store, type, snapshot) {
     this._init(store, type);
-    var data = this._recordToData(store, type, record);
-    return this.get('db').rel.del(this.getRecordTypeName(type), data)
-      .then(extractDeleteRecord);
+    let data = this._recordToData(store, type, snapshot);
+
+    //Delete the children
+    let deletedChildren = [];
+    snapshot.eachRelationship(function(name, descriptor) {
+      if(descriptor.kind === 'hasMany' && descriptor.options['cascadeDestroy']) {
+        snapshot.record.get(name).then(function(childRecords) {
```

Comment:
You are right, I did mess something up. What you have is pretty close, it should be fixed in the latest commit.